src/libraries/System.Net.Security/src/System/Net/Security/MD4.cs

```diff
@@ -0,0 +1,230 @@
+        private const int S31 = 3;
+        private const int S32 = 9;
+        private const int S33 = 11;
+        private const int S34 = 15;
+
+        internal static void HashData(ReadOnlySpan<byte> source, Span<byte> destination)
+        {
+            Debug.Assert(destination.Length == 128 >> 3);
+            Span<byte> buffer = stackalloc byte[64];
+            Span<uint> state = stackalloc uint[4];
+            Span<uint> count = stackalloc uint[2];
+
+            // the initialize our context
+            count[0] = 0;
+            count[1] = 0;
+            state[0] = 0x67452301;
+            state[1] = 0xefcdab89;
+            state[2] = 0x98badcfe;
+            state[3] = 0x10325476;
+            // Zeroize sensitive information
+            buffer.Fill(0);
+
+            HashCore(source, state, count, buffer);
+
+            /* Save number of bits */
+            Span<byte> bits = stackalloc byte[8];
+            Encode(bits, count);
+
+            /* Pad out to 56 mod 64. */
+            uint index = ((count[0] >> 3) & 0x3f);
+            int padLen = (int)((index < 56) ? (56 - index) : (120 - index));
+            Span<byte> padding = stackalloc byte[padLen];
+            padding[0] = 0x80;
+            HashCore(padding, state, count, buffer);
+
+            /* Append length (before padding) */
+            HashCore(bits, state, count, buffer);
+
+            /* Write state to destination */
+            Encode(destination, state);
+        }
+
+        private static void HashCore(ReadOnlySpan<byte> input, Span<uint> state, Span<uint> count, Span<byte> buffer)
+        {
+            /* Compute number of bytes mod 64 */
+            int index = (int)((count[0] >> 3) & 0x3F);
+            /* Update number of bits */
+            count[0] += (uint)(input.Length << 3);
+            if (count[0] < (input.Length << 3))
+                count[1]++;
+            count[1] += (uint)(input.Length >> 29);
+
+            int partLen = 64 - index;
+            int i = 0;
+            /* Transform as many times as possible. */
+            if (input.Length >= partLen)
+            {
+                BlockCopy(input, 0, buffer, index, partLen);
+                MD4Transform(state, buffer);
+
+                for (i = partLen; i + 63 < input.Length; i += 64)
+                {
+                    MD4Transform(state, input.Slice(i));
+                }
+
+                index = 0;
+            }
+
+            /* Buffer remaining input */
+            input.Slice(i).CopyTo(buffer.Slice(index));
+        }
+
+        //--- private methods ---------------------------------------------------
+
+        private static void BlockCopy(ReadOnlySpan<byte> source, int srcOffset, Span<byte> destination, int dstOffset, int count)
+        {
+            for (int srcIndex = srcOffset, dstIndex = dstOffset; srcIndex < srcOffset + count; srcIndex++, dstIndex++)
+            {
+                destination[dstIndex] = source[srcIndex];
+            }
```

Comment:
If this method is still needed, the body can be replaced with

```suggestion
            source.Slice(srcOffset, count).CopyTo(destination.Slice(dstOffset);
```

Since there's only one callsite, the method could be inlined/removed.